corehq/apps/userreports/views.py

```diff
@@ -73,7 +73,7 @@
 def get_datasource_config_or_404(config_id, domain):
     try:
         return get_datasource_config(config_id, domain)
-    except DataSourceConfigurationNotFoundError:
+    except (DataSourceConfigurationNotFoundError, BadSpecError):
         raise Http404
```

Comment:
I was going to write that it's not clear to me that we should treat a bad spec as a 404 and propose using a `messages.error` and redirect kind of thing, but then I saw the underlying error was a missing data source.

It kind of feels to me like the cleaner thing to do would be to raise a (new) `DataSourceNotFoundError` (which is probably a subclass of `BadSpecError`) in the code that is being called and then trap that here. 

(not saying it would obviously be worth the effort to change it at this point, but i think that would be more correct)